packages/react/src/components/Dialog/index.tsx

```diff
@@ -28,196 +28,159 @@ export interface DialogProps extends React.HTMLAttributes<HTMLDivElement> {
   portal?: React.RefObject<HTMLElement> | HTMLElement;
 }
 
-interface DialogState {
-  isolator?: AriaIsolate;
-}
-
 const isEscape = (event: KeyboardEvent) =>
   event.key === 'Escape' || event.key === 'Esc' || event.keyCode === 27;
 
-const noop = () => {
-  //not empty
-};
-
-export default class Dialog extends React.Component<DialogProps, DialogState> {
-  static defaultProps = {
-    onClose: noop,
-    forceAction: false,
-    closeButtonText: 'Close'
-  };
-
-  private element: HTMLDivElement | null;
-  private heading: HTMLHeadingElement | null;
-  private headingId: string = nextId('dialog-title-');
-
-  constructor(props: DialogProps) {
-    super(props);
-
-    this.close = this.close.bind(this);
-    this.focusHeading = this.focusHeading.bind(this);
-    this.handleClickOutside = this.handleClickOutside.bind(this);
-    this.handleEscape = this.handleEscape.bind(this);
-    this.state = {};
-  }
-
-  componentDidMount() {
-    if (this.props.show) {
-      this.attachEventListeners();
-      this.attachIsolator(() => setTimeout(this.focusHeading));
-    }
-  }
-
-  componentWillUnmount() {
-    const { isolator } = this.state;
-    isolator?.deactivate();
-    this.removeEventListeners();
-  }
-
-  componentDidUpdate(prevProps: DialogProps) {
-    if (!prevProps.show && this.props.show) {
-      this.attachIsolator(this.focusHeading);
-      this.attachEventListeners();
-    } else if (prevProps.show && !this.props.show) {
-      this.removeEventListeners();
-      this.close();
-    }
-  }
-
-  private attachIsolator(done: () => void) {
-    this.setState(
-      {
-        isolator: new AriaIsolate(this.element as HTMLElement)
-      },
-      done
-    );
-  }
-
-  render() {
-    const {
-      dialogRef,
-      forceAction,
+const Dialog = forwardRef<HTMLDivElement, DialogProps>(
```

Comment:
Most of the primary work here was to convert Dialog to a functional component instead of class one to use the focus trap hook.